common/app/components/Nav/Avatar-Points-Nav-Item.jsx

```diff
@@ -4,21 +4,35 @@ import { Link } from 'react-router';
 // this is separated out to prevent react bootstrap's
 // NavBar from injecting unknown props to the li component
 export default function AvatarPointsNavItem({ picture, points, username }) {
+  const profileUsername = username ?
+  <span className='hidden-md hidden-lg'> { username } </span> :
+  <span className='hidden-md hidden-lg'> Signing you in </span>;
+
+  const placeholderImg = 'https://github.com/FreeCodeCamp/assets/blob/master' +
+    '/assets/logos/fcc_puck600.png';
+
+  const profileImg = picture ?
+    (<img
+      className='profile-picture float-right'
+      src={ picture }
+    />) :
+    (<img
+      className='profile-picture float-right'
+      src={ placeholderImg }
+    />);
+
   return (
     <li
       className='avatar-points'
       key='user'
       >
       <Link to='/settings'>
         <span className='brownie-points-nav'>
-          <span className='hidden-md hidden-lg'> { username } </span>
+          { profileUsername }
```

Comment:
Stay DRY by moving the element creation back down here use the ternary above to determine the content of that element

```jsx
<span>{ content}</span>
```